cmd/minishift/cmd/root.go

```diff
@@ -203,6 +200,36 @@ func initConfig() {
 	setupViper()
 }
 
+func initializeProfile() string {
+	var (
+		profileName string
+		err         error
+	)
+
+	for i, arg := range os.Args {
+		if arg == "--profile" {
+			profileName = os.Args[i+1]
+			break
+		}
+	}
+
+	// We need to initialize allinstance config as the active profile information present in it.
+	ensureAllInstanceConfigPath(constants.AllInstanceConfigPath)
+	minishiftConfig.AllInstancesConfig, err = minishiftConfig.NewAllInstancesConfig(constants.AllInstanceConfigPath)
+	if err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error creating all instance config: %s", err.Error()))
+	}
+	activeProfile := profileActions.GetActiveProfile()
+
+	if profileName != "" {
+		return profileName
+	} else if activeProfile != "" {
+		return activeProfile
+	} else {
```

Comment:
no need for else?
```
	if profileName != "" {
		return profileName
	} else if activeProfile != "" {
		return activeProfile
	}
	return ""
}

Reply:
@gbraad @budhrg I prefer the code to be like this as it is very clear and readable. Not sure why you want to get rid of the last else. It will work for sure. But in the current form it is more readable.

Reply:
I disagree

return statements should be clearly visible. Either at the top as bailout when doing requirements and variable checking, and at the end when a result is given.

There are a lot of unnecessary uses of `else` that make the code hard to read and reason about.

Reply:
```
func max(a int, b int) int
{
    if (a > b) {
        return a;
    } else {
        return b;
    }
}
```

```
func max(a int, b int) int
{
    if (a > b) {
        return a;
    }
    return b;
}
```

Reply:
https://eslint.org/docs/rules/no-else-return
https://www.cachequality.com/docs/rules/unnecessary-else